[RFC] Remove instances of TRUE/FALSE macro

steveno opened the pull request:
memory.c
os_unix.c
path.c

steveno committed:
Remove instances of TRUE/FALSE macro

memory.c
os_unix.c
path.c

bfredl commented on src/nvim/os_unix.c:
```diff
@@ -681,6 +681,6 @@ static int have_dollars(int num, char_u **file)
     if (vim_strchr(file[i], '$') != NULL)
-      return TRUE;
-  return FALSE;
+      return true;
```
change return type as well?

bfredl replied:
stupid github won't let me put a comment there, but looks like `have_wildcard` could be converted as well.

steveno replied:
Good catch! Updated.

bfredl commented on src/nvim/memory.c:
```diff
@@ -496,7 +496,7 @@ void free_all_mem(void)
 
   /* Close all tabs and windows.  Reset 'equalalways' to avoid redraws. */
-  p_ea = FALSE;
+  p_ea = false;
```
For a global variable, I think it's better changing all occurrences at once instead of just those that happen to be in a single file, whenever reasonable. As `p_ea` and `p_acd` are not so heavily used, I would suggest fixing the rest of them (incl the definition type) in this PR.

steveno replied:
When I pushed this commit I was unsure how to handle that. On a second, closer, examination, I think I should be able to fix this as you've suggested. I'll fix and re-push.

fmoralesc commented on src/nvim/path.c:
```diff
@@ -1156,10 +1156,10 @@ int gen_expand_wildcards(int num_pat, char_u **pat, int *num_file,
-          recursive = TRUE;
-          did_expand_in_path = TRUE;
+          recursive = true;
+          did_expand_in_path = true;
```
lint is complaining for the lack of braces around the else clause just below this line.

steveno replied:
I see it. I will add that in.

fmoralesc commented on src/nvim/path.c:
```diff
@@ -1393,10 +1393,10 @@ void simplify_filename(char_u *filename)
             if (os_fileinfo((char *)filename, &file_info)) {
-              do_strip = TRUE;
+              do_strip = true;
             }
```
`}` -> `} else {`, and then close the else clause braces.

steveno replied:
Ditto for this one.

steveno commented:
After making the changes to the global variables as requested I get the two following warnings. I'm not sure how to proceed. Suggestions?

![screenshot from 2015-08-31 21 57 03](https://cloud.githubusercontent.com/assets/116258/9594626/514916d8-502b-11e5-89ea-642e41d90ecc.png)

bfredl commented:
That would be fixed by changing `(int *)` to `(bool *)`.

But I am a little bit worried by `*(int *)varp = value;` in the beginnig of `set_bool_option`, won't this be undefined behaviour when `varp` really points to a `bool` ? But we have changed boolean options to `EXTERN bool p_option;` before, so I will look for discussion in the earlier PRs.

bfredl commented:
No discussion in #645 or #1227/#1711 . There was some discussion in #1345 about endianness problems, but it didn't touch this line. But `showoneopt` uses `*(bool*`)varp` for all boolean options. I wonder if someone has tested neovim recently on a big-endian system?

steveno commented:
I updated based upon the suggestions from @bfredl (thanks!).

I also fixed two more functions in `option.c` that was returning TRUE/FALSE to return a `bool` now: `file_ff_differs` and `can_bs`.

`make` and `make test` worked without error locally.

steveno commented:
The Travis failure, oddly enough, is over a line I didn't touch.

bfredl commented:
Yes, the linter suppresses existing errors, but depends on surrounding lines to recognize an error as preexisting. If it is simple enough (like in this case), just fix it as well.

steveno committed:
merge changes

fmoralesc commented on src/nvim/ex_cmds.c:
```diff
@@ -4429,16 +4429,15 @@ void ex_global(exarg_T *eap)
       smsg(_("Pattern found in every line: %s"), pat);
-    else
+    } else {
       smsg(_("Pattern not found: %s"), pat);
```
Missing } after this line, I think.

steveno commented:
I f**ked my branch and can't fix it, and the build is failing because I'm short a `}` from a botched merge due to conflicts. 

I'll be on vacation for a week. Any help while I'm gone would be appreciated; I **really** want to get this merged ASAP. It's starting to feel like a rabbit hole I can't escape from.

fmoralesc commented:
We've all been there, don't worry. I'll take a look locally.

fmoralesc committed:
fixup: appease clint

fmoralesc commented:
@steveno Just opened a PR in your fork that should fix the current issues. Let's cross fingers ;)

steveno committed:
Merge pull request #1 from fmoralesc/rmTF1

fixup: appease clint

steveno commented:
@fmoralesc thank you so much!

This hopefully is ready to merge!

bfredl commented:
Merged, thanks!

For a future PR, please rebase your changes on top of master, instead of merging master changes into your PR. It makes it easier to do the final review/commit to master.

bfredl commented:
Also, while unrealistic for every function this kind of refactor touches, in the cases you anyway need to reformat the function signature, consider changing to doxygen-style function documentation http://neovim.io/develop/style-guide.xml?showone=Function_Comments#Function_Comments
